src/ucp/core/ucp_ep.c

```diff
@@ -803,6 +804,10 @@ static void ucp_destroyed_ep_pending_purge(uct_pending_req_t *self, void *arg)
 
 void ucp_ep_destroy_internal(ucp_ep_h ep)
 {
+    if (--ep->ref_cnt != 0) {
```

Comment:
This way we leave ucp_ep visible to keepalive_pregress but not checkable (ep_check returns timeout). Hence warnings in ucp_ep_do_keepalive().

Reply:
> This way we leave ucp_ep visible to keepalive_progress but not checkable (ep_check returns timeout). Hence warnings in ucp_ep_do_keepalive().

@Artemy-Mellanox good catch, thank you!
moved this check to `ucp_ep_destroy_base()` where UCP EP is about to be removed